src/compiler/emitter.ts

```diff
@@ -3055,17 +3052,7 @@ export function createPrinter(printerOptions: PrinterOptions = {}, handlers: Pri
             const text = getLiteralTextOfNode(expression as LiteralExpression, /*neverAsciiEscape*/ true, /*jsxAttributeEscape*/ false);
             // If the number will be printed verbatim and it doesn't already contain a dot or an exponent indicator, add one
             // if the expression doesn't have any comments that will be emitted.
-            return !(expression.numericLiteralFlags & TokenFlags.WithSpecifier)
-                && !stringContains(text, tokenToString(SyntaxKind.DotToken)!)
-                && !stringContains(text, String.fromCharCode(CharacterCodes.E))
-                && !stringContains(text, String.fromCharCode(CharacterCodes.e));
-        }
-        else if (isAccessExpression(expression)) {
-            // check if constant enum value is integer
-            const constantValue = getConstantValue(expression);
-            // isFinite handles cases when constantValue is undefined
-            return typeof constantValue === "number" && isFinite(constantValue)
-                && Math.floor(constantValue) === constantValue;
+            return !isNumericLiteralTextSafeForPropertyAccess(expression, text);

```

Comment:
I need to recheck when this line is actually hit - if the team's preference would be to just replace the "dot dot" strategy with parentheses then likely this whole function here could be just removed. Thoughts?